ui/src/components/Exchange/Exchange.tsx

```diff
@@ -1,21 +1,47 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import { Switch, Route, useRouteMatch } from 'react-router-dom'
 
+import { useParty, useLedger, useStreamFetchByKey } from '@daml/react'
+import { useWellKnownParties } from '@daml/dabl-react'
+import {
+    Exchange as ExchangeTemplate,
+    ExchangeInvitation
+} from '@daml.js/da-marketplace/lib/Marketplace/Exchange'
+
 import Page from '../common/Page'
 import WelcomeHeader from '../common/WelcomeHeader'
+import { wrapDamlTuple } from '../common/damlTypes'
 
 import ExchangeSideNav from './ExchangeSideNav'
 import MarketPairs from './MarketPairs'
 import CreateMarket from './CreateMarket'
+import ExchangeParticipants from './ExchangeParticipants'
 
 type Props = {
     onLogout: () => void;
 }
 
 const Exchange: React.FC<Props> = ({ onLogout }) => {
     const { path, url } = useRouteMatch();
+    const operator = useWellKnownParties().userAdminParty;
+    const user = useParty();
+    const ledger = useLedger();
+
+    const key = () => wrapDamlTuple([operator, user]);
+    const exchangeContract = useStreamFetchByKey(ExchangeTemplate, key, [operator, user]).contract;
+    const exchangeInvite = useStreamFetchByKey(ExchangeInvitation, key, [operator, user]).contract;
 
-    const sideNav = <ExchangeSideNav url={url}/>;
+    useEffect(() => {
+        if (exchangeInvite) {
```

Comment:
This could be something that we handle on an exchange bot but as discussed it should be ok here for now